tests/driver_lis2dh12/main.c

```diff
@@ -11,170 +11,682 @@
+        /* enables interrupt on Y-axis below the threshold value */
+        params_int2.int_config = LIS2DH12_INT_CFG_YLIE;
+        params_int2.int_duration = 1;
+        params_int2.cb = lis2dh12_int_cb;
+        params_int2.arg = &line2;
     }
 
-    /* create and set the interrupt params */
-    if (gpio_is_valid(lis2dh12_params[0].int2_pin)) {
-        lis2dh12_int_params_t params_int2 = {
-            .int_type = LIS2DH12_INT_TYPE_I2_IA2,
-            .int_config = LIS2DH12_INT_CFG_YLIE,
-            .int_threshold = 31,
-            .int_duration = 1,
-            .cb = lis2dh12_int_cb,
-            .arg = &ctx[1],
-        };
-        lis2dh12_set_int(&dev, &params_int2, LIS2DH12_INT2);
+    if (gpio_init_int(pin1, GPIO_IN, GPIO_RISING, lis2dh12_int_cb, &line1)) {
+        DEBUG("[lis_init]: INT1 failed\n");
+    }
+    else {
+        DEBUG("[lis_init]: INT1 done\n");
     }
-#endif
 
-    while (1) {
+    if (gpio_init_int(pin2, GPIO_IN, GPIO_RISING, lis2dh12_int_cb, &line2)) {
+        DEBUG("[lis_init]: INT2 failed\n");
+    }
+    else {
+        DEBUG("[lis_init]: INT2 done\n");
+    }
+#endif /* MODULE_LIS2DH12_INT */
+
+    /* enable FIFO */
+    lis2dh12_set_fifo(&dev, &fifo_cfg);
+
+    /* enable click detection */
+    lis2dh12_set_click(&dev, &click_cfg);
 
+    /* set default shock value */
+    shock_thold = THOLD_SHOCK_MILLIG_DEFAULT;
+
+    /* read registers to reset device */
+    lis2dh12_read_click_src(&dev, &click_src_reg);
+    lis2dh12_read_reference(&dev, &reference_value);
 #ifdef MODULE_LIS2DH12_INT
-        if (xtimer_mutex_lock_timeout(&isr_mtx, DELAY) == 0) {
-            flags = isr_flags;
-            isr_flags = 0;
+    lis2dh12_read_int_src(&dev, &int1_src, 1);
+#endif /* MODULE_LIS2DH12_INT */
+}
+
+#ifdef MODULE_LIS2DH12_INT
+void* lis2dh12_test_process(void* arg) {
+    (void) arg;
+    while (1) {
+        /* start processing */
+        DEBUG("[Process]: start process\n");
+
+        /* read FIFO_src before getting data */
+        LIS2DH12_FIFO_SRC_REG_t fifo_src;
+        lis2dh12_read_fifo_src(&dev, &fifo_src);
+        DEBUG("[Process]: FIFO SRC 0x%x\n", fifo_src.reg);
+        DEBUG("[Process]: WTM %x, OVRN %d, EMPTY %d, FSS %d\n", fifo_src.bit.WTM,
+                                fifo_src.bit.OVRN_FIFO, fifo_src.bit.EMPTY, fifo_src.bit.FSS);
+
+        /* get fifo data */
+        uint8_t number_read = lis2dh12_read_fifo_data(&dev, data_fifo, NUM_FIFO_VALUES);
+
+        /* read FIFO_src after getting data */
+        lis2dh12_read_fifo_src(&dev, &fifo_src);
+        DEBUG("[Process]: FIFO SRC 0x%x\n", fifo_src.reg);
+        DEBUG("[Process]: WTM %x, OVRN %d, EMPTY %d, FSS %d\n", fifo_src.bit.WTM,
+                                fifo_src.bit.OVRN_FIFO, fifo_src.bit.EMPTY, fifo_src.bit.FSS);
+
+        /* display FIFO data */
+        if (ENABLE_DEBUG) {
+            for (int i = 0; i < number_read; i++){
+                printf("[Process]: X_%2d  %d\n", i ,data_fifo[i].X_AXIS);
```

Comment:
Should be more readable this way.

```suggestion
                printf("[Process]: X[%2d]  %d\n",i , data_fifo[i].X_AXIS);
```
(same below) 

or even better, why not use the same format as `list read`

    [0] X: -0.800  Y: -0.032  Z:  0.640
    [1] X: -0.800  Y: -0.064  Z:  0.640